src/main/java/org/elasticsearch/index/mapper/geo/GeoShapeFieldMapper.java

```diff
@@ -273,40 +277,38 @@ public void merge(Mapper mergeWith, MergeContext mergeContext) throws MergeMappi
             return;
         }
         final GeoShapeFieldMapper fieldMergeWith = (GeoShapeFieldMapper) mergeWith;
-        if (!mergeContext.mergeFlags().simulate()) {
-            final PrefixTreeStrategy mergeWithStrategy = fieldMergeWith.defaultStrategy;
+        final PrefixTreeStrategy mergeWithStrategy = fieldMergeWith.defaultStrategy;
 
-            // prevent user from changing strategies
-            if (!(this.defaultStrategy.getClass().equals(mergeWithStrategy.getClass()))) {
-                mergeContext.addConflict("mapper [" + names.fullName() + "] has different strategy");
-            }
+        // prevent user from changing strategies
+        if (!(this.defaultStrategy.getClass().equals(mergeWithStrategy.getClass()))) {
+            mergeContext.addConflict("mapper [" + names.fullName() + "] has different strategy");
+        }
 
-            final SpatialPrefixTree grid = this.defaultStrategy.getGrid();
-            final SpatialPrefixTree mergeGrid = mergeWithStrategy.getGrid();
+        final SpatialPrefixTree grid = this.defaultStrategy.getGrid();
+        final SpatialPrefixTree mergeGrid = mergeWithStrategy.getGrid();
 
-            // prevent user from changing trees (changes encoding)
-            if (!grid.getClass().equals(mergeGrid.getClass())) {
-                mergeContext.addConflict("mapper [" + names.fullName() + "] has different tree");
-            }
+        // prevent user from changing trees (changes encoding)
+        if (!grid.getClass().equals(mergeGrid.getClass())) {
+            mergeContext.addConflict("mapper [" + names.fullName() + "] has different tree");
+        }
 
-            // TODO we should allow this, but at the moment levels is used to build bookkeeping variables
-            // in lucene's SpatialPrefixTree implementations, need a patch to correct that first
-            if (grid.getMaxLevels() != mergeGrid.getMaxLevels()) {
-                mergeContext.addConflict("mapper [" + names.fullName() + "] has different tree_levels or precision");
-            }
+        // TODO we should allow this, but at the moment levels is used to build bookkeeping variables
+        // in lucene's SpatialPrefixTree implementations, need a patch to correct that first
+        if (grid.getMaxLevels() != mergeGrid.getMaxLevels()) {
+            mergeContext.addConflict("mapper [" + names.fullName() + "] has different tree_levels or precision");
+        }
 
-            // bail if there were merge conflicts
-            if (mergeContext.hasConflicts()) {
-                return;
-            }
+        // bail if there were merge conflicts
+        if (mergeContext.hasConflicts()) {
```

Comment:
Or maybe you meant to add the simulate check here? Otherwise this members would be modified when simulating (simulation is like the "validation" pass in mappings merging)